### PR TITLE
Fix Header (+ Refactor)

### PR DESCRIPTION
**Pull request by martinezguillaume**

### CODE REVIEW

**Comment by iRoachie** [comment]
Compared to the picture we have in the docs

File: src/header/__tests__/Header.js [hunk 1]
```diff
@@ -34,13 +35,15 @@ describe('Header Component', () => {
       </Header>
     );
 
-    expect(component.find('Button').length).toBe(2);
+    expect(component.length).toBe(1);
+    expect(toJson(component)).toMatchSnapshot();
```
Comment (Gregjarvez) [inline src/header/__tests__/Header.js]: Should we be snapshotting here ?. Can we test the actually functionality describe by the test case 😉
`should render multiple children when passed in`

`expect(component.children().length).toBe(2)`

or more explicitly 

`expect(component.find('Button').length).toBe(2)`

same with the other cases :)

**Comment by martinezguillaume** [comment]
@iRoachie It's fixed
@Gregjarvez I revert the test changes

**Comment by iRoachie** [comment]
@martinezguillaume Mind doing the typescript definitions or you want me to do em?

**Comment by martinezguillaume** [comment]
@iroachie I’ll do it, I forgot 🙂

**Comment by xavier-villelegier** [comment]
Ping @martinezguillaume 🔔

**Pull request merged**